website/src/_posts/2017-08-28-timely-processing.md

```diff
@@ -428,8 +481,38 @@ new DoFn<Event, EnrichedEvent>() {
 ```
 
 ```py
-# State and timers are not yet supported in Beam's Python SDK.
-# Follow https://issues.apache.org/jira/browse/BEAM-2687 for updates.
+class StatefulBufferingFn(beam.DoFn):
+  …
+
+  STALE_TIMER = TimerSpec('stale', TimeDomain.REAL_TIME)
+
+  MAX_BUFFER_DURATION = 1
+
+  def process(self, element,
+              w=beam.DoFn.WindowParam,
+              buffer_state=beam.DoFn.StateParam(BUFFER_STATE),
+              count_state=beam.DoFn.StateParam(COUNT_STATE),
+              expiry_timer=beam.DoFn.TimerParam(EXPIRY_TIMER),
+              stale_timer=beam.DoFn.TimerParam(STALE_TIMER)):
+
+    if count_state.read() == 0:
+      # In Python, real-time timers are set on absolute time.
+      stale_timer.set(time.time() + StatefulBufferingFn.MAX_BUFFER_DURATION)
```

Comment:
This comment only makes sense if you are familiar with the Java API. Perhaps rephrase to

```
# We set an absolute timestamp here (not an offset like in the Java SDK)
```